### DISCUSSION TITLE
MacOS - dotenv not found after installation

### DISCUSSION THREAD

**Discussion by Guneetsinghtuli** [osm-search/Nominatim#3005] [Q&A]
I have been trying to install Nominatim locally using this [Documentation](https://nominatim.org/release-docs/develop/admin/Installation/). After successful compilation when I tried running `nominatim`  on the command line. I get an `ModuleNotFoundError: No module named 'dotenv'`. 
Even after installing [Python-dotenv](https://github.com/theskumar/python-dotenv). I still am facing the same error, Would be really helpful if someone could help me out with this.
![Screenshot 2023-03-11 at 12 43 49 AM](https://user-images.githubusercontent.com/48324492/224405683-bca675e9-6ff0-4f9d-bcab-657e05a20e15.png)

**Comment by mtmail** [comment]
So this command `python3 -c "from dotenv import dotenv_values"` should return no error. How did you install dotenv, with pip3? Is pip3 in the same path as python3 (`which pip3; which python3`)? Which version of Nominatim are you using, master from github or a stable release (https://github.com/osm-search/Nominatim/releases) ? Which version of MacOS and which architecture (Intel vs Apple Silicon), I think `uname -a` prints that information?

I seem to remember somebody installed nominatim successful on MacOS but can't find the thread. Nominatim isn't officially supported on MacOS but could work.

Unrelated: If possible copy&paste the text instead of a screenshot. That will help other users looking for the same error message in the future.

**Comment by Guneetsinghtuli** [comment]
```
guneetsingh@GUNEETs-MacBook-Air ~ % python3 -c "from dotenv import dotenv_values"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'dotenv'
guneetsingh@GUNEETs-MacBook-Air ~ % python3 --version
Python 3.9.0
```

Path of pip3 and python3
`guneetsingh@GUNEETs-MacBook-Air build % which pip3; which python3
/usr/local/bin/pip3
/usr/local/bin/python3`


I tried to install dotenv with pip3 using the command `pip3 install python-dotenv` as it was mentioned in the docs.
I am using the Nominatim from the master branch and my architecture is intel based MacOS 
the output from the command `uname -a` is this

`Darwin GUNEETs-MacBook-Air.local 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64 x86_64`

Maybe the next documentation i am gonna write will be on installing it on MacOS

**Comment by lonvia** [comment]
Please check again that you really have installed `python-dotenv`. There is another package called `dotenv` which is completely different.

If that is okay, then there is an issue with however pip is set up. I recommend you switch to using a [virtualenv](https://virtualenv.pypa.io/en/latest/) for the Python libraries.
